### DISCUSSION TITLE
Type caster for reference_wrapper

### DISCUSSION THREAD

**Discussion by akoen** [wjakob/nanobind#384] [General]
I'd like to bind a c++ function that takes a vector of wrapped objects _without_ calling their copy constructor and explicitly disallowing a null argument.

I can of course use pointers e.g. `std::vector<const Type*>` but nanobind will happily allow a `None` argument which I have to explicitly handle internally.

`pybind11` [defines](https://github.com/pybind/pybind11/blob/7d538a42750c8580eeaac10e505840a3694b04c8/include/pybind11/cast.h#L57) a type caster for std::reference_wrapper that does (I believe) what I'm looking. Is there similar support in `nanobind`?

**Comment by wjakob** [comment]
No such type caster currently exists, but you are welcome to provide a PR that adds this functionality.
